designer/client/__tests__/ComponentTypeEdit.jest.tsx

```diff
@@ -0,0 +1,370 @@
+import React, { useReducer } from "react";
+import { render } from "@testing-library/react";
+import ComponentTypeEdit from "../ComponentTypeEdit";
+import {
+  ComponentContext,
+  componentReducer,
+  initComponentState,
+} from "../reducers/component/componentReducer";
+import { DataContext } from "../context";
+import { initI18n } from "../i18n";
+import { Data } from "@xgovformbuilder/model";
+
+initI18n();
+
+describe("ComponentTypeEdit", () => {
+  let mockData: Data;
+
+  const RenderWithContext = ({ children, stateProps = {} }) => {
+    const [state, dispatch] = useReducer(
+      componentReducer,
+      initComponentState({
+        ...stateProps,
+      })
+    );
+
+    return (
+      <DataContext.Provider value={{ data: mockData, save: jest.fn() }}>
+        <ComponentContext.Provider value={{ state, dispatch }}>
+          {children}
+        </ComponentContext.Provider>
+      </DataContext.Provider>
```

Comment:
if you don't need to rely on dispatches for your tests (like click -> expect something changed) you can remove this useReducer and do

```suggestion
  const RenderWithContext = ({ children, stateProps = {} }) => {

    return (
      <DataContext.Provider value={{ data: mockData, save: jest.fn() }}>
        <ComponentContext.Provider value={{state: stateProps, dispatch: jest.fn()}}>
          {children}
        </ComponentContext.Provider>
      </DataContext.Provider>
```

Reply:
Thanks @jenbutongit, happy to remove it but maybe it is better leaving it because it is work done for when we start deprecating lab tests and move behaviour tests here?